ceph/exporter.go

```diff
@@ -86,6 +93,84 @@ func (exporter *Exporter) cephVersionCmd() []byte {
 	return cmd
 }
 
+func CephVersionsCmd() ([]byte, error) {
+	// Ceph versions
+	cmd, err := json.Marshal(map[string]interface{}{
+		"prefix": "versions",
+		"format": "json",
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal ceph versions command: %s", err)
+	}
+	return cmd, nil
+}
+
+func ParseCephVersions(buf []byte) (map[string]map[string]float64, error) {
+	// Rather than a dedicated type, have dynamic daemons and versions
+	// {"daemon": {"version1": 123, "version2": 234}}
+	parsed, err := gabs.ParseJSON(buf)
+	if err != nil {
+		return nil, err
+	}
+
+	parsedMap, err := parsed.ChildrenMap()
+	if err != nil {
+		return nil, err
+	}
+
+	versions := make(map[string]map[string]float64)
+	for daemonKey, innerObj := range parsedMap {
+		// Read each daemon, and overall counts
+		versionMap, err := innerObj.ChildrenMap()
+		if err == gabs.ErrNotObj {
+			continue
+		} else if err != nil {
+			return nil, err
+		}
+
+		versions[daemonKey] = make(map[string]float64)
+		for version, countContainer := range versionMap {
+			count, ok := countContainer.Data().(float64)
+			if ok {
+				versions[daemonKey][version] = count
+			}
+		}
+	}
+
+	return versions, nil
+}
+
+func (exporter *Exporter) setRbdMirror() error {
+
+	cmd, err := CephVersionsCmd()
+	if err != nil {
+		exporter.Logger.WithError(err).Panic("failed to marshal ceph versions command")
+	}
+
+	buf, _, err := exporter.Conn.MonCommand(cmd)
+	if err != nil {
+		exporter.Logger.WithError(err).WithField(
+			"args", string(cmd),
+		).Error("error executing mon command")
+		return err
+	}
+
+	versions, err := ParseCephVersions(buf)
+	if err != nil {
+		return err
+	}
+
+	// check to see if rbd-mirror is in ceph version output and not empty
+	if _, exists := versions["rbd-mirror"]; exists {
+		version := versions["rbd-mirror"]
+		if len(version) != 0 {
+			exporter.RbdMirror = true
+		}
+	}
```

Comment:
Because the length of a `nil` map is `0`, you could shorten this to:
```go
if len(versions["rbd-mirror"]) > 0 {
    exporter.RbdMirror = true
}
```
if you wanted.

Reply:
I shall do that!